### ISSUE TITLE
While pulling v2 tags, panic: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by stevenschlansker** [docker/docker#15923]
This error message is both unhelpful as to pointing to the actual cause, and causes the entire Docker daemon to crash when an upstream registry misbehaves:

```
ERRO Error from V2 registry: strconv.ParseInt: parsing "": invalid syntax 
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x8 pc=0x641d2c]

goroutine 1830 [running]:
net/http.(*chunkWriter).flush(0xc209eca2a0)
        /usr/local/go/src/net/http/server.go:283 +0x5c
net/http.(*response).Flush(0xc209eca280)
        /usr/local/go/src/net/http/server.go:1048 +0x66
github.com/docker/docker/pkg/ioutils.(*WriteFlusher).Write(0xc2099e29c0, 0xc2096e9320, 0xad, 0x108, 0xad, 0x0, 0x0)
        /go/src/github.com/docker/docker/pkg/ioutils/writeflusher.go:21 +0x145
github.com/docker/docker/pkg/progressreader.(*Config).Read(0xc2084b8b60, 0xc20a1de000, 0x8000, 0x8000, 0x8000, 0x7f6cdc49cf80, 0xc20802a0d0)
        /go/src/github.com/docker/docker/pkg/progressreader/progressreader.go:37 +0x2e5
io.Copy(0x7f6cdc4a12a8, 0xc20803bad0, 0x7f6cd82f1b38, 0xc2084b8b60, 0x0, 0x0, 0x0)
        /usr/local/go/src/io/io.go:362 +0x1f6
github.com/docker/docker/graph.func·008(0xc20a174d80, 0x0, 0x0)
        /go/src/github.com/docker/docker/graph/pull.go:602 +0xd42
github.com/docker/docker/graph.func·009(0xc20a174d80)
        /go/src/github.com/docker/docker/graph/pull.go:626 +0x2f
created by github.com/docker/docker/graph.(*TagStore).pullV2Tag
        /go/src/github.com/docker/docker/graph/pull.go:627 +0x2671
```

```
root@mesos-slave2-qa-uswest2:~# docker info
Containers: 207
Images: 565
Storage Driver: overlay
 Backing Filesystem: extfs
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 4.0.4
Operating System: Ubuntu 14.04.2 LTS
CPUs: 8
Total Memory: 29.45 GiB
Name: mesos-slave2-qa-uswest2
ID: 4UUZ:F25M:YWO7:MUWG:H4AV:5LKT:T2MG:6I3Q:Q7MU:2G2C:2OFE:DJUZ
```

```
root@mesos-slave2-qa-uswest2:~# docker version
Client version: 1.7.1
Client API version: 1.19
Go version (client): go1.4.2
Git commit (client): 786b29d
OS/Arch (client): linux/amd64
Server version: 1.7.1
Server API version: 1.19
Go version (server): go1.4.2
Git commit (server): 786b29d
OS/Arch (server): linux/amd64
```

**Comment by thaJeztah** [comment]
@stevenschlansker have you been able to reproduce this on docker 1.8.x or 1.9?

**Comment by stevenschlansker** [comment]
We're on 1.8.3, and I think it does not affect that version.  We stopped seeing these panics around our upgrade time window.

**Comment by thaJeztah** [comment]
@stevenschlansker thanks! IIRC, some changes were made to fix this in 1.8, so I'll close this issue, but don't hesitate to comment if you see this happening again. Thanks for reporting!

**Comment by kkdevops** [comment]
anybody know the cause of this docker crash on 1.7?
